Issue 5588 - add migrator that returns removed default values to old blocks

myroslavve opened the pull request:
# Description
Some default attributes were removed in https://github.com/maxi-blocks/maxi-blocks/pull/5579.
This PR adds a migrator that adds those default attribute values where necessary.
<!---
Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
--->

Fixes #5588 

# How Has This Been Tested?
Checked that FAQ-PRO-1 looks right, and other patterns don't break.
<!---
Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.
--->

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [x] Checked that FAQ-PRO-1 looks right, and other patterns don't break.

# Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings/errors
-   [x] I have added/updated tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced migration functionality for typography unit defaults.
  - Added support for saving deprecated blocks with optional attribute ignoring.

- **Improvements**
  - Enhanced block registration to handle deprecated blocks more effectively.
  - Updated state management to better handle deprecated block saving.

- **Bug Fixes**
  - Fixed issues with deprecated block attribute saving logic.

- **Developer Enhancements**
  - Improved internal block migration processes and state management for deprecated blocks.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

myroslavve committed:
Add migrator that returns removed default values to old blocks

elzadj reviewed (changes requested):
Buttons are ok now, but the Pane still has the undefined unit:

![Screenshot from 2024-05-21 16-30-12](https://github.com/maxi-blocks/maxi-blocks/assets/2401618/5382f356-7a85-459f-b1fe-c3d7278888a4)

Olekrut commented:
![image](https://github.com/maxi-blocks/maxi-blocks/assets/88893350/a1b476fa-af53-41ce-a13c-6778e0520385)
were added default bottom-gap  to some of the old patterns
INL_01
INL-06

dashaversha commented:
https://jam.dev/c/09836576-93e9-4223-be3b-1da553d35413
INL-05
the way I can fix this

myroslavve committed:
Fix accordion not migrating

myroslavve committed:
Fix migrators not running after `uniqueIDMigrator`

New uniqueID was passed to migrators' save and they didn't generate the same save content because of that, so migrators weren't run

myroslavve committed:
Fix for custom formats

myroslavve committed:
Merge branch 'master' into issue-5588

myroslavve committed:
Add new versions of maxi to migrator

myroslavve commented:
> ![image](https://private-user-images.githubusercontent.com/88893350/332454329-a1b476fa-af53-41ce-a13c-6778e0520385.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4MDc0NjQsIm5iZiI6MTcxNjgwNzE2NCwicGF0aCI6Ii84ODg5MzM1MC8zMzI0NTQzMjktYTFiNDc2ZmEtYWY1My00MWNlLWExM2MtNjc3OGUwNTIwMzg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI3VDEwNTI0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzN2MyZjBmZWQwNjQwNzU4MDAzZjNhNWZjOWE1NWE2ZTRkMWNiN2RlZDY3ZTI5ZDU4N2JlMWIwMzg1MTg2MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AXAr1FXgZTgT_1TUCQXx32BTtpshfQEITrwRvt4OcOU) were added default bottom-gap to some of the old patterns INL_01 INL-06 ![image](https://private-user-images.githubusercontent.com/88893350/332456472-ef179b82-a6f5-4571-907f-2306aa981a32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4MDc0NjQsIm5iZiI6MTcxNjgwNzE2NCwicGF0aCI6Ii84ODg5MzM1MC8zMzI0NTY0NzItZWYxNzliODItYTZmNS00NTcxLTkwN2YtMjMwNmFhOTgxYTMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI3VDEwNTI0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxYjRjM2VmMDExMzBlNzdmNTc4N2RhYzEwMzk1NWU0YzJjYjRmOTZmMjMxYmYzOTc5NmUzMTIzYjg0MDRiMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CT6LCJFtC9X6zcqdiMlbe8URSquswDkuIRHssx5pOIo) probably because it was created before we add bottom-gap to Maxi

Fixed.

Olekrut reviewed (approved):
hard to test it, because we have a lot of combination for that, but ,looks like everything is ok) 
good job)))
